### DISCUSSION TITLE
How to use a model I've trained on another system.

### DISCUSSION THREAD

**Discussion by Paasu** [neuropoly/axondeepseg#292] [Help needed] [answered]
I have been training a model at home and I would like to use the model to segment images on a computer at my lab? How would I proceed doing that? Also, if I want to train 300 epochs but don't want to leave my computer on for 50 hours straight can I just train 50 epochs every night? Would it be the same or does it make it different?

**Comment by alexfoias** [comment]
~~@Paasu i’ll send you the details by email~~
Please disregard my previous message. @mathieuboudreau  could you please help him ?

**Comment by vasudev-sharma** [comment, marked as answer]
1. If you want to reuse use model, then you could save you model via this command once it has been trained `model.save("model_name.h5")` and to reload this saved model you could do this `model.load_model("model_name.h5")`. For more information to save and reload model you could refer this [blog post](https://machinelearningmastery.com/save-load-keras-deep-learning-models/)
2. As per you  second question is considered, you can train your models for small number of epochs daily using **checkpoints callback** to the model . So you have to checkpoint your model and reload this checkpoint model later to train it for further epochs. You can refer this [link](https://towardsdatascience.com/checkpointing-deep-learning-models-in-keras-a652570b8de6) for more information. Also, it would be same as training straight for 300 epochs, for your case

**Comment by vasudev-sharma** [comment]
Also note in AxonDeepSeg we have implemented this behaviour 
1. For loading the models [here](https://github.com/neuropoly/axondeepseg/blob/715f0e61cf17313155b20d8cf6d1e44549a6c222/AxonDeepSeg/train_network.py#L272)
2. For using checkpoint callbacks [here](https://github.com/neuropoly/axondeepseg/blob/715f0e61cf17313155b20d8cf6d1e44549a6c222/AxonDeepSeg/train_network.py#L221)

**Comment by Paasu** [comment]
Sorry for the late response, but if I am correct, you have already implemented it and I can already do what I originally planned without any changes correct?
I will be able to do 50 epochs per night as planned?

**Comment by vasudev-sharma** [comment]
@Paasu, since we were having the GPU resources we didn't felt the need to load the models using the checkpoints. However, for your purpose, you can add   `model.load_weights(filepath_loss)` or 
`model.load_weights(filepath_acc)  ` (depending upon whether you want to optimize the loss or accuracy of your mode) to the `train_network.py` script at [here](https://github.com/neuropoly/axondeepseg/blob/73f0c90b882ea645acac8a069a0e5010a96c52f8/AxonDeepSeg/train_network.py#L248).  Remember that, that if your goal is to let's say train for 500 epochs with the checkpoint of 50 epochs each, you need to make some changes in the code.
1. First of all, in Model Checkpoint callback, you need to set period = 50 [here](https://github.com/neuropoly/axondeepseg/blob/73f0c90b882ea645acac8a069a0e5010a96c52f8/AxonDeepSeg/train_network.py#L233) and [here](https://github.com/neuropoly/axondeepseg/blob/73f0c90b882ea645acac8a069a0e5010a96c52f8/AxonDeepSeg/train_network.py#L246).
2. Second, you need to reduce the number of epochs for training after each checkpoint is completed. For instance, after training for the first 50 epochs, you need to set the training epochs to 450 ( 500 - 50), because if you don't do this you will training for a longer number of epochs ( greater than 500).

PS: let me know if something is not clear.
